`to_json`: `timedelta_mode` and `temporal_mode` break documented behavior

### Initial Checks

- [x] I confirm that I'm using Pydantic V2

### Description

`pydantic_core.to_json` (and `to_jsonable_python`) documents `timedelta_mode` and `temporal_mode` as independent parameters governing disjoint type sets:

> - `timedelta_mode`: How to serialize `timedelta` objects, either `'iso8601'` or `'float'`.
> - `temporal_mode`: How to serialize datetime-like objects (`datetime`, `date`, `time`), either `'iso8601'`, `'seconds'`, or `'milliseconds'`.

(This is what both the [API docs](https://pydantic.dev/docs/validation/latest/api/pydantic-core/pydantic_core/) and the shipped `_pydantic_core.pyi` type stub say.)

Given that contract, `to_json(..., timedelta_mode="float", temporal_mode="iso8601")` should serialize `datetime`/`date`/`time` as ISO 8601 strings and `timedelta` as a float. Instead, `datetime`/`date`/`time` are **also** coerced to unix-timestamp floats, `timedelta_mode="float"` silently overrides `temporal_mode="iso8601"` for all temporal types.

There is no combination of the two parameters that yields ISO datetimes with float-second timedeltas, even though the documented parameters imply there should be.

**Root cause / regression.** This regressed in [#1743](https://github.com/pydantic/pydantic-core/pull/1743). Before that PR, `SerializationConfig` had a dedicated `timedelta_mode: TimedeltaMode` field and `to_json` had no `temporal_mode` parameter, so `timedelta_mode` was genuinely independent. #1743 replaced the dedicated field with a single unified `temporal_mode` and retrofitted `timedelta_mode` as a legacy alias that maps into it.

Either the docs/`.pyi` should be corrected to state that `timedelta_mode` is a deprecated alias that affects all temporal types, or (preferably) `timedelta` should retain an independent setting so the documented behavior holds. Personally I have a requirement to be able to set the `timedelta` format independent of the other `datetime` types.

### Example Code

```Python
import datetime
import pydantic_core

payload = {
    "dt": datetime.datetime(2026, 7, 9, 12, 0, 0),
    "d": datetime.date(2026, 7, 9),
    "t": datetime.time(12, 0, 0),
    "td": datetime.timedelta(hours=1, minutes=30),
}

# Per the docs: temporal_mode governs dt/d/t; timedelta_mode governs td.
actual = pydantic_core.to_json(
    payload, timedelta_mode="float", temporal_mode="iso8601"
).decode()

expected = '{"dt":"2026-07-09T12:00:00","d":"2026-07-09","t":"12:00:00","td":5400.0}'

print("actual  :", actual)
print("expected:", expected)
assert actual == expected, "timedelta_mode='float' wrongly coerced datetime/date/time to floats"
# AssertionError: actual == {"dt":1783598400.0,"d":1783555200.0,"t":43200.0,"td":5400.0}
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.13.4
        pydantic-core version: 2.46.4
          pydantic-core build: profile=release pgo=false
               python version: 3.11.14 (main, Oct 10 2025, 12:48:11) [Clang 20.1.4 ]
                     platform: Linux-6.17.0-35-generic-x86_64-with-glibc2.39
             related packages: typing_extensions-4.16.0 mypy-2.2.0 email-validator-2.3.0 pydantic-settings-2.14.2 pydantic-extra-types-2.11.1
                       commit: unknown
```

--- a/pydantic-core/python/pydantic_core/_pydantic_core.pyi
+++ b/pydantic-core/python/pydantic_core/_pydantic_core.pyi
@@ -449,10 +449,31 @@ def to_json(
         by_alias: Whether to use the alias names of fields.
         exclude_none: Whether to exclude fields that have a value of `None`.
         round_trip: Whether to enable serialization and validation round-trip support.
-        timedelta_mode: How to serialize `timedelta` objects, either `'iso8601'` or `'float'`.
-        temporal_mode: How to serialize datetime-like objects (`datetime`, `date`, `time`), either `'iso8601'`, `'seconds'`, or `'milliseconds'`.
-            `iso8601` returns an ISO 8601 string; `seconds` returns the Unix timestamp in seconds as a float; `milliseconds` returns the Unix timestamp in milliseconds as a float.
+        timedelta_mode: The format of serialized timedeltas. Accepts the string values of `'iso8601'` and `'float'`.
 
+            - `'iso8601'` will serialize timedeltas to [ISO 8601 text format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
+            - `'float'` will serialize timedeltas to the total number of seconds.
+
+            /// version-changed | v2.12
+            It is now recommended to use the `temporal_mode` argument. `timedelta_mode` will be deprecated in v3.
+            ///
+        temporal_mode: The format of serialized temporal types from the [`datetime`][] module. This includes:
+
+            - [`datetime.datetime`][]
+            - [`datetime.date`][]
+            - [`datetime.time`][]
+            - [`datetime.timedelta`][]
+
+            Can be one of:
+
+            - `'iso8601'` will serialize date-like types to [ISO 8601 text format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
+            - `'milliseconds'` will serialize date-like types to a floating point number of milliseconds since the epoch.
+            - `'seconds'` will serialize date-like types to a floating point number of seconds since the epoch.
+
+            /// version-added | v2.12
+            This argument replaces `timedelta_mode`, which will be deprecated in v3. `temporal_mode` adds more
+            configurability for the other temporal types. It takes precedence over `timedelta_mode`.
+            ///
         bytes_mode: How to serialize `bytes` objects, either `'utf8'`, `'base64'`, or `'hex'`.
         inf_nan_mode: How to serialize `Infinity`, `-Infinity` and `NaN` values, either `'null'`, `'constants'`, or `'strings'`.
         serialize_unknown: Attempt to serialize unknown types, `str(value)` will be used, if that fails
@@ -534,10 +555,31 @@ def to_jsonable_python(
         by_alias: Whether to use the alias names of fields.
         exclude_none: Whether to exclude fields that have a value of `None`.
         round_trip: Whether to enable serialization and validation round-trip support.
-        timedelta_mode: How to serialize `timedelta` objects, either `'iso8601'` or `'float'`.
-        temporal_mode: How to serialize datetime-like objects (`datetime`, `date`, `time`), either `'iso8601'`, `'seconds'`, or `'milliseconds'`.
-            `iso8601` returns an ISO 8601 string; `seconds` returns the Unix timestamp in seconds as a float; `milliseconds` returns the Unix timestamp in milliseconds as a float.
+        timedelta_mode: The format of serialized timedeltas. Accepts the string values of `'iso8601'` and `'float'`.
 
+            - `'iso8601'` will serialize timedeltas to [ISO 8601 text format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
+            - `'float'` will serialize timedeltas to the total number of seconds.
+
+            /// version-changed | v2.12
+            It is now recommended to use the `temporal_mode` argument. `timedelta_mode` will be deprecated in v3.
+            ///
+        temporal_mode: The format of serialized temporal types from the [`datetime`][] module. This includes:
+
+            - [`datetime.datetime`][]
+            - [`datetime.date`][]
+            - [`datetime.time`][]
+            - [`datetime.timedelta`][]
+
+            Can be one of:
+
+            - `'iso8601'` will serialize date-like types to [ISO 8601 text format](https://en.wikipedia.org/wiki/ISO_8601#Durations).
+            - `'milliseconds'` will serialize date-like types to a floating point number of milliseconds since the epoch.
+            - `'seconds'` will serialize date-like types to a floating point number of seconds since the epoch.
+
+            /// version-added | v2.12
+            This argument replaces `timedelta_mode`, which will be deprecated in v3. `temporal_mode` adds more
+            configurability for the other temporal types. It takes precedence over `timedelta_mode`.
+            ///
         bytes_mode: How to serialize `bytes` objects, either `'utf8'`, `'base64'`, or `'hex'`.
         inf_nan_mode: How to serialize `Infinity`, `-Infinity` and `NaN` values, either `'null'`, `'constants'`, or `'strings'`.
         serialize_unknown: Attempt to serialize unknown types, `str(value)` will be used, if that fails

--- a/pydantic/config.py
+++ b/pydantic/config.py
@@ -595,6 +595,7 @@ class ConfigDict(TypedDict, total=False):
     /// version-added | v2.12
     This setting replaces [`ser_json_timedelta`][pydantic.config.ConfigDict.ser_json_timedelta],
     which will be deprecated in v3. `ser_json_temporal` adds more configurability for the other temporal types.
+    It takes precedence over `ser_json_timedelta`.
     ///
     """
 

